runtime/parachains/src/ump.rs

```diff
@@ -540,7 +544,9 @@ impl<T: Config> Pallet<T> {
 						let _ = queue_cache.consume_front::<T>(dispatchee);
 					},
 					Err((id, required)) => {
-						if required.any_gt(config.ump_max_individual_weight) {
+						let is_under_limit = Overweight::<T>::count() < MAX_OVERWEIGHT_MESSAGES;
+						weight_used.saturating_accrue(T::DbWeight::get().reads(1));
```

Comment:
Stuff like this doesn't work any more with 2D weights. It needs a proper benchmark.